Make Webpacker the default JavaScript compiler for Rails 6

dhh opened the pull request:
Make Webpacker the default Javascript compiler for Rails 6 with the following changes:

- Webpacker gem is installed by default and webpacker:install is run by the Rails application generator.
- Action Cable channel generators will create ES6 stubs rather than use CoffeeScript
- Active Storage, Action Cable, Turbolinks, and Rails-UJS is loaded by a new application.js pack in app/javascript by default (unless any of the frameworks have been opted out).
- Active Storage, Action Cable, Turbolinks, and Rails-UJS npm modules are automatically listed as dependencies in the default package.json.
- All the JavaScript-related auxiliaries for Sprockets, like compression and uglifying, is no longer configured or included by default.
- No JavaScript stubs will be created by default when using the scaffold generators any more.

Commit:
Use Webpacker by default on new apps

Commit:
Stop including coffee-rails by default

Commit:
Drop using a js_compressor by default

Commit:
Drop extra test for coffeescript inclusion by default

Commit:
Stick with skip_javascript to signify skipping webpack

Commit:
Don't install a JS runtime by default any more

Commit:
app/javascript will be the new default directory for JS

Commit:
Make it clear that this is just for configuring the default Webpack framework setup now

Commit:
Start using the Webpack tag in the default layout

Commit:
Irrelevant test

Commit:
jQuery is long gone

Commit:
Stop having asset pipeline compile default application.js

Commit:
Add rails-ujs by default to the Webpack setup

Commit:
Add Active Storage JavaScript to application.js pack by default

Commit:
Consistent quoting

Commit:
Add Turbolinks to default pack

Commit:
Add Action Cable to default pack

Need some work on how to set the global consumer that channels will
work with. @javan?

yhirano55 commented on railties/lib/rails/generators/app_base.rb:
```diff
@@ -327,59 +324,26 @@ def convert_database_option_for_jruby
-        else
-          GemfileEntry.new "mini_racer", nil, comment, { platforms: :ruby }, true
+          [ GemfileEntry.version("turbolinks", "~> 5",
+             "Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks") ]
```
Turbolinks is added on package.json, so it looks unnecessary to add turbolinks to Gemfile.

yhirano55 replied:
I see. Thanks. https://github.com/turbolinks/turbolinks-rails

dhh commented:
There’s a Ruby and a JS component. Both are needed. 

> On Jun 7, 2018, at 23:23, yhirano55 <notifications@github.com> wrote:
> 
> @yhirano55 commented on this pull request.
> 
> In railties/lib/rails/generators/app_base.rb:
> 
> > -          end
> -
> -          gems
> -        end
> -      end
> -
> -      def javascript_runtime_gemfile_entry
> -        comment = "See https://github.com/rails/execjs#readme for more supported runtimes"
> -        if defined?(JRUBY_VERSION)
> -          GemfileEntry.version "therubyrhino", nil, comment
> -        elsif RUBY_PLATFORM =~ /mingw|mswin/
> -          GemfileEntry.version "duktape", nil, comment
> -        else
> -          GemfileEntry.new "mini_racer", nil, comment, { platforms: :ruby }, true
> +          [ GemfileEntry.version("turbolinks", "~> 5",
> +             "Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks") ]
> Turbolinks is added on package.json, so it looks unnecessary to add turbolinks to Gemfile.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

Commit:
Require all channels by default and use a separate consumer stub

rainerborene commented:
@dhh any reason to keep the `javascript` folder name in singular? why not following the pattern of plural name like controllers, models and views?

dhh commented:
Because JavaScript is a language, not a classification of a domain type,
like controllers or views or models. JavaScript has all those things.
Saying "JavaScripts" imply that it's a collection of isolated, independent
files, when it's not. It's the part of the application written in
JavaScript.

On Fri, Jun 15, 2018 at 6:18 PM, Rainer Borene <notifications@github.com>
wrote:

> @dhh <https://github.com/dhh> any reason to keep the javascript folder
> name in singular? why not following the pattern of plural name like
> controllers, models and views?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/rails/rails/pull/33079#issuecomment-397671410>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAAKtdNf_Y4N_9wLEZE3ZnZR08tvrNhiks5t895pgaJpZM4UegdM>
> .
>

Commit:
Channel generator now targets Webpack style

Commit:
Merge branch 'master' into webpacker-default

Commit:
Update task docs to match new generator style

Commit:
Use uniform import style

dhh commented:
No, Sprockets is still to be used for SCSS (and other CSS processors) as
well as copying all other static assets.

On Mon, Jun 25, 2018 at 3:25 PM, Rainer Borene <notifications@github.com>
wrote:

> @dhh <https://github.com/dhh> that means that sprockets will no longer be
> a dependency on Rails?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/rails/rails/pull/33079#issuecomment-399950394>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAAKtUSOtzstvc81TfoLi1IQVNJijU_Zks5uAOTngaJpZM4UegdM>
> .
>

Commit:
Drop the JS assets generator

It was barely helpful as it was. It’s no longer helpful in a Webpacked
world. Sayonara!

Commit:
Add app/javascript to the stats directories

Commit:
Simpler import style

Which match the other imports.

dwightwatson commented:
Out of curiousity, what is the argument to continue using Sprockets for CSS/static assets when Webpacker supports them by default out of the box?

adrianthedev commented:
My `javascripts` folder already has the `images`, `js`, `packs`, `stylesheets` folders inside. Would be nice to use the old `assets` folder instead of the current `javascripts` to keep all assets.

Commit:
Address test failures from dropping JS compilation (and compression)

rainerborene commented:
> No, Sprockets is still to be used for SCSS (and other CSS processors) as well as copying all other static assets.

@dhh I see your point. But it's inevitable that these JavaScript packages will make use of their own internal styling (like React or something). So, you'll end up with styles in two separate places and from different "compilers". That could become a mess, don't you think?

dhh commented:
I don’t think it’s a problem. You can have Webpack do your CSS as well, if you like. But the default path is not having to muck with that. 

> On Jun 27, 2018, at 19:08, Rainer Borene <notifications@github.com> wrote:
> 
> No, Sprockets is still to be used for SCSS (and other CSS processors) as well as copying all other static assets.
> 
> @dhh I see your point. But it's inevitable that these JavaScript packages will make use of their own internal styling (like React or something). So, you'll end up with styles in two separate places and from different "compilers". That could become a mess, don't you think?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

frodsan commented:
@adrianthedev I think you can change the default path in `.webpacker.yml`:

```diff
default: &default
-  source_path: app/javascript
+ source_path: app/assets
```

adrianthedev commented:
For sure I can do that @frodsan. But by doing it that it would change the default "rails" way.

This is the turning point when we can have that default path in `assets` and all generators and other packages point to that. 
Again, making it the rails way (and the consistent way) of keeping everything (js, images, css, etc.) in the `assets` folder and **giving the developer the choice of using Webpack or sprockets** for his CSS.
It seems as the perfect setup.

Commit:
Merge branch 'master' into webpacker-default

coorasse commented:
Keep supporting both sprockets and webpacker seems a bad idea: how do I explain to new developers that javascripts are managed by webpacker and stylesheets by sprockets? Seems a bit crazy to me.
I would rather make a cut, take a clear decision, and switch to webpacker completely, rename the `javascripts` folder to `webpacker` (not `assets` to simplify migration process) and have the courage to embrace webpack 100%.

dhh commented:
If you can't explain to developers that JavaScript is managed by one process and stylesheets by another, I think maybe you're moving too fast in your educational program with these developers. Understanding that JavaScript is separate from stylesheets seems like a pretty fundamental thing to start with before diving into full application development.

But you're also free to change your curriculum to focus on Webpacker. There is support for Webpacker handling other assets in addition to JavaScript, but it's not going to be the default in Rails 6.

dhh commented:
This branch is now nearly ready to merge, but there are a fair amount of railties tests that need to be updated to match the new generation defaults. If anyone on this thread is interested in helping getting this PR to master, I'd much appreciate some assistance updating these railties tests. You can start a new PR based off the webpacker-default branch.

Once this is merged to master, I'm going to proceed with the reveal of the new Rails 6.0 framework that depends on these changes 😄

ahorek commented:
```
sprockets are unmaintained at least for a year
sass unmaintained
sassc unmaintained
```

but sprockets is still a hard runtime dependency in rails, I think it should be optional.

My concern is that migration of an existing app from sprockets to full stack webpacker isn't easy, it requires many, many changes. It may even discourage people from upgrading if it would be the only option. We also decided to move to webpacker because working with sprockets was really painful. I support the idea that it should be default even for sass/css.

But step by step, let's finish this PR first, merge it and discuss later.

rafaelfranca commented:
> sprockets are unmaintained at least for a year

Who said that? We just released a new 4.0 beta 28 days ago.

ahorek commented:
well, sprockets 4 is in beta since February 18, 2016 and I don't believe it'll ever be released in stable. Thanks for your work on it, but AFAIK you're the only maintainer.

dhh commented:
This is unproductive. Please refrain from any further distraction on this thread that’s unrelated to the scope of this PR: Making Webpacker the default JavaScript compiler for Rails 6.

If you wish to have a conversation beyond that scope, please take it to the mailing list or elsewhere.

> On Jul 17, 2018, at 20:52, ahorek <notifications@github.com> wrote:
> 
> well, sprockets 4 is in beta since February 18, 2016 and I don't believe it'll ever be released in stable. Thanks for your work on it, but AFAIK you're the only maintainer.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

ttanimichi committed:
webpacker-default: Modify `AssetsGeneratorTest`

Before:

```
$ bin/test test/generators/assets_generator_test.rb
Run options: --seed 46201

F

Failure:
AssetsGeneratorTest#test_assets [/Users/ttanimichi/ghq/github.com/ttanimichi/rails/railties/test/generators/assets_generator_test.rb:12]:
Expected file "app/assets/javascripts/posts.js" to exist, but does not

bin/test /Users/ttanimichi/ghq/github.com/ttanimichi/rails/railties/test/generators/assets_generator_test.rb:10

.

Finished in 0.031343s, 63.8101 runs/s, 95.7152 assertions/s.
2 runs, 3 assertions, 1 failures, 0 errors, 0 skips
```

After:

```
$ bin/test test/generators/assets_generator_test.rb
Run options: --seed 43571

..

Finished in 0.030370s, 65.8545 runs/s, 65.8545 assertions/s.
2 runs, 2 assertions, 0 failures, 0 errors, 0 skips
```

ttanimichi committed:
webpacker-default: Modify `ChannelGeneratorTest`

Before:

```
$ bin/test test/generators/channel_generator_test.rb
Run options: --seed 8986

.F

Failure:
ChannelGeneratorTest#test_channel_with_multiple_actions_is_created [/Users/ttanimichi/ghq/github.com/ttanimichi/rails/railties/test/generators/channel_generator_test.rb:43]:
Expected file "app/assets/javascripts/channels/chat.js" to exist, but does not

bin/test /Users/ttanimichi/ghq/github.com/ttanimichi/rails/railties/test/generators/channel_generator_test.rb:34

.F

Failure:
ChannelGeneratorTest#test_channel_is_created [/Users/ttanimichi/ghq/github.com/ttanimichi/rails/railties/test/generators/channel_generator_test.rb:29]:
Expected file "app/assets/javascripts/channels/chat.js" to exist, but does not

bin/test /Users/ttanimichi/ghq/github.com/ttanimichi/rails/railties/test/generators/channel_generator_test.rb:22

E

Error:
ChannelGeneratorTest#test_cable_js_is_created_if_not_present_already:
Errno::ENOENT: No such file or directory @ apply2files - /Users/ttanimichi/ghq/github.com/ttanimichi/rails/railties/test/fixtures/tmp/app/assets/javascripts/cable.js

bin/test /Users/ttanimichi/ghq/github.com/ttanimichi/rails/railties/test/generators/channel_generator_test.rb:60

F

Failure:
ChannelGeneratorTest#test_channel_suffix_is_not_duplicated [/Users/ttanimichi/ghq/github.com/ttanimichi/rails/railties/test/generators/channel_generator_test.rb:87]:
Expected file "app/assets/javascripts/channels/chat.js" to exist, but does not

bin/test /Users/ttanimichi/ghq/github.com/ttanimichi/rails/railties/test/generators/channel_generator_test.rb:80

F

Failure:
ChannelGeneratorTest#test_channel_on_revoke [/Users/ttanimichi/ghq/github.com/ttanimichi/rails/railties/test/generators/channel_generator_test.rb:77]:
Expected file "app/assets/javascripts/cable.js" to exist, but does not

bin/test /Users/ttanimichi/ghq/github.com/ttanimichi/rails/railties/test/generators/channel_generator_test.rb:68

Finished in 0.064384s, 108.7227 runs/s, 481.4861 assertions/s.
7 runs, 31 assertions, 4 failures, 1 errors, 0 skips
```

After:

```
$ bin/test test/generators/channel_generator_test.rb
Run options: --seed 44857

.......

Finished in 0.060243s, 116.1961 runs/s, 697.1764 assertions/s.
7 runs, 42 assertions, 0 failures, 0 errors, 0 skips
```

guilleiguaran commented:
Update: after of merging the PRs of @ttanimichi  we still having failures in the next files:

```
Failed in:
  test/application/rake_test.rb
  test/generators/controller_generator_test.rb
  test/generators/scaffold_controller_generator_test.rb
  test/generators/app_generator_test.rb
  test/generators/plugin_generator_test.rb
  test/generators/scaffold_generator_test.rb
```

dhh commented:
Thanks a lot for helping with the test updates, @ttanimichi 🙏

ttanimichi committed:
webpacker-default: Modify `ControllerGeneratorTest`

The JS assets generator was dropped. ref. https://github.com/rails/rails/commit/46215b179483d3e4d264555f5a4952f43eb8142a

ttanimichi commented:
My pleasure. I think webpacker is a good compromise to manage modern JS in Rails.

kaspth committed:
Fix shared generator tests.

kaspth commented:
Tried to reduce more of the failures. Several tests spit this out:

```ruby
kaspers-macbook-pro:railties kasperhansen$ bin/test test/generators/app_generator_test.rb -n /test_generator_does_not_generate_active_storage_contents_if_skip_active_record_is_given/
Run options: -n /test_generator_does_not_generate_active_storage_contents_if_skip_active_record_is_given/ --seed 31939

# Running:

rails aborted!
Don't know how to build task 'webpacker:install' (see --tasks)
/Users/kasperhansen/code/rails/railties/lib/rails/commands/rake/rake_command.rb:23:in `block in perform'
/Users/kasperhansen/code/rails/railties/lib/rails/commands/rake/rake_command.rb:20:in `perform'
/Users/kasperhansen/code/rails/railties/lib/rails/command.rb:48:in `invoke'
/Users/kasperhansen/code/rails/railties/lib/rails/commands.rb:18:in `<main>'
/Users/kasperhansen/code/rails/activesupport/lib/active_support/dependencies.rb:287:in `block in require'
/Users/kasperhansen/code/rails/activesupport/lib/active_support/dependencies.rb:253:in `load_dependency'
/Users/kasperhansen/code/rails/activesupport/lib/active_support/dependencies.rb:287:in `require'
bin/rails:4:in `<main>'
(See full trace by running task with --trace)
.

Finished in 2.971897s, 0.3365 runs/s, 7.0662 assertions/s.
1 runs, 21 assertions, 0 failures, 0 errors, 0 skips
```

So it looks like the `TestApp::Application` doesn't have access to `webpacker:install`. I tried adding `webpacker` to the Gemfile with a require in `railties/test/abstract_unit.rb`, but then the test would just hang 😕

`isolation/abstract_unit.rb` likely also needs to require webpacker somehow (in the same vein that it does with Turbolinks).

ttanimichi commented:
@kaspth Maybe that's because `ENV['BUNDLE_GEMFILE']` is set. We have discussed it in [this thread](https://github.com/rails/rails/pull/32780#discussion_r185307068).

Commit:
Merge branch 'master' into webpacker-default

Commit:
Revert "Simpler import style". It's currently failing with an error of "TypeError: undefined is not an object (evaluating '__WEBPACK_IMPORTED_MODULE_2_activestorage___default.a.start')". Waiting for @javan to have a look.

This reverts commit 5d3ebb71059f635d3756cbda4ab9752027e09256.

Commit:
Merge branch 'master' into webpacker-default

Commit:
require webpacker in test app

kaspth committed:
Merge pull request #33619 from lsylvester/require-webpacker-in-test-app

require webpacker in test app

lsylvester commented:
@dhh Is it possible to get a release of webpacker that includes rails/webpacker@8ca372dc500d16a2343aaade7646647aac6925b3. I think that this is required for the `webpacker:install` commands to run correctly within the generators.

I can currently manually adding it to the test app in https://github.com/rails/rails/pull/33622/files#diff-6b400e02467a1f4e0c2c60863c87b927R471 but this should not be requires and is causing some issues with the generator tests.

lsylvester commented:
I have found another place where the build can hang. In `bin/rails app:update` it waits for user input  to confirm to overwrite the development.rb and production.rb as they are different because of the code that web packer injects into them. If we include the webpacker config in the templates in the railties then then it can continue without hanging waiting for user input.

n-rodriguez commented on railties/lib/rails/generators/rails/app/templates/package.json.tt:
```diff
@@ -1,5 +1,10 @@
   "private": true,
-  "dependencies": {}
+  "dependencies": {
+    "rails-ujs": "<%= Rails::VERSION::STRING %>"<% unless options[:skip_turbolinks] %>,
```
Hi there! I think there's something wrong here... `"rails-ujs" unless options[:skip_turbolinks]`, etc...

georgeclaghorn replied:
This is correct: we’re skipping the *next* dependency.

lsylvester commented:
I am having issues with the build timing out for my branches that are trying to provide fixes for this. 

The new generator step of `webpacker:install` is causing some of the generator tests (I am looking at `bin/test test/generators/plugin_generator_test.rb  -v` currently) to take significantly longer than before (from less than a second to around 20 seconds). From what I can tell, most of this time is running `yarn add` commands and deleting files. 

I am trying to work out ways to speed up the `yarn add` step, but I have had no success so far. If anyone has ideas on how to do this that would be really useful.

Otherwise, would it be acceptable to skip javascript on all the tests for generators  that aren't explicitly trying to do anything with the javascript?

The only other option that I can think of is to split the railties tests on travis into different jobs so that each job can run under 50 minutes.

dhh commented:
I think it's perfectly fine to turn off javascript generation for tests
that do not explicitly seek to test that.

On Mon, Aug 20, 2018 at 12:18 AM, lsylvester <notifications@github.com>
wrote:

> I am having issues with the build timing out for my branches that are
> trying to provide fixes for this.
>
> The new generator step of webpacker:install is causing some of the
> generator tests (I am looking at bin/test test/generators/plugin_generator_test.rb
> -v currently) to take significantly longer than before (from less than a
> second to around 20 seconds). From what I can tell, most of this time is
> running yarn add commands and deleting files.
>
> I am trying to work out ways to speed up the yarn add step, but I have
> had no success so far. If anyone has ideas on how to do this that would be
> really useful.
>
> Otherwise, would it be acceptable to skip javascript on all the tests for
> generators that aren't explicitly trying to do anything with the javascript?
>
> The only other option that I can think of is to split the railties tests
> on travis into different jobs so that each job can run under 50 minutes.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/rails/rails/pull/33079#issuecomment-414219978>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAAKtVUPAqHf8ttLlXBeYSlqsQroZjavks5uSmKxgaJpZM4UegdM>
> .
>

lsylvester commented:
#33640 which adds web packer to the gentile without handing/timing out the build is ready for review.

lsylvester commented:
Don't anyone have any guidance on what the js support for a newly generated plugin should have?

Going forwards, plugins will have to publish the javascript via npm so that they can be used by apps using webpacker. So I think that they will need to have at a minimum a `package.json` and a `app/javascript/%name%/index.js`. 

I think that it would be good if they could be generated with the ability to compile the `app/javascript` to `app/assets/javascripts` so that they could ship the javascript for both webpacker and sprockets applications, but I am not sure what the most appropriate tool to do this would be. Active Storage seems to be using [rollup](https://github.com/rollup/rollup) to do this. Would generating a plugin with a minimal rollup config be a sensible default?

dhh commented:
Anyone interested in picking up the work on the testing side and push this over the finish line? I have that new framework sitting all pretty waiting for this to land in master 😄

Commit:
Merge branch 'master' into webpacker-default

lsylvester commented:
@dhh Did you have any thoughts on the work that was done in #33640 for this? I think from a build speed point of view (which is there main problem it is facing) I think the most important thing would be to replace the `yarn add` in web packer which direct modification of `package.json` and a skip-able `yarn install`. This would allow for the tests that check the generated files but don't need to actually run the javascript skip interacting with yarn at all, which is the main bottleneck.

lsylvester committed:
Add webpacker without making the build hang/timeout. (#33640)

* use yarn workspaces to allow for installing unreleased packages and only generate js/bootsnap when required

* no longer need to have webpacker in env templates as webpacker moved this config to yml file

dhh commented:
@lsylvester I hadn't seen that. Merged it in! Do see my note about hardcoding the framework version, though. Fine skipping yarn install for the tests 👍

Commit:
Merge branch 'master' into webpacker-default

sharang-d committed:
Fix rubocop violation

rafaelfranca committed:
Merge pull request #33931 from sharang-d/fix-rubocop-issue

Fix rubocop issue in the webpacker-default branch

lsylvester committed:
Got the test passing for the running scaffold

lsylvester committed:
update expected lines of code

lsylvester committed:
update middleware tests to account for webpacker

rafaelfranca committed:
Merge pull request #33965 from lsylvester/weback-default-test-fixes

More test fixes for webpack-default

lsylvester commented:
#34009 Now has all the tests passing for this branch 🎉 

These is still some more things to clean up - like the removing the hard coded package version numbers and figuring out the js support for engines.

The failure on `test/application/rake_test.rb` that is on this branch I didn't get of the last PR so I think it might be related to cached node_modules or something. The error is `DRb::DRbConnError` which is not super helpful - it is likely it was an error in an template as they can't be marshalled. I am also working on a fix for that.

/cc @dhh

lsylvester committed:
disable js in plugins be default to get the tests passing (#34009)

Commit:
Merge branch 'master' into webpacker-default

dhh commented:
@lsylvester Can you clear the codeclimate report issues?

consti committed:
clear codeclimate report issues

lsylvester commented:
@dhh looks like @consti beat me to it.

georgeclaghorn committed:
Merge pull request #34024 from consti/webpacker-default

Clear codeclimate report issues

dhh commented:
There’s still a bunch of failures on the railties suite. Also, for now, we should have an open ended version bind to the npm dependencies. Just the latest released version.

prashuchaudhary commented:
@TomaszGrudzinski I don't think there is any problem with that commit.  I patched the same thing in a sample app and it works.

 It might be that for some reason definition of `active_support/hash_with_indifferent_access.rb` is conflicting when `rails/webpacker` is requiring `active_support/core_ext/hash/indifferent_access`

lsylvester commented:
I have a fix for the the `except` issue for `HashWithIndiferentAccess` on #34031. That branch is against master as it is not specific to webpacker. 

Also #34030 should help with the `DRb` errors being reported as it will allow error messages raised from templates be seen with the real error messages.

It would be good to get those merged and master merged back into this branch.

Also #34011 will remove the hard coding of the package versions and is the latest patch release for the package. I am not sure about allowing any of the latest packages as at some point the packages would become incompatible and generating a app in an old version of rails would result in it being in a broken state.

@dhh Are you happy constrained to latest patch version, or do you want it fully opened up. I think at some point we need to have the gems being able to control which version of the package is used - but I haven't had a chance to work on that yet.

Commit:
Merge branch 'master' into webpacker-default

Commit:
Anything newer than currently released is good

dhh commented:
@lsylvester Awesome. I've merged everything and pushed the latest. Let's see if this makes the CI happy and we can finally merge!

I'm a fan of open dependency versioning unless you know that there's a specific incompatibility you're defending against. So leaving them open.

Commit:
Use Webpacker development version during development of Rails

Commit:
Edge should get development webpacker as well

Commit:
Add changelog entry for Webpacker change

partounian commented:
I can't wait to upgrade from Rails 4 to Rails 6, this ticket is making me hyped

n-rodriguez commented:
> I can't wait to upgrade from Rails 4 to Rails 6, this ticket is making me hyped

maybe you should migrate to Rails 5 before :)

shashiprab commented:
@dhh keeping my eyes peeled for the promised new JS framework for Rails 6.0 reveal! (Haven't found anything satisfying so far with either React, or Stimulus or Ember - hope its a new "Rails Way" framework)

dhh commented:
It's been revealed! https://www.youtube.com/watch?v=HJZ9TnKrt7Q

There's another Rails 6 reveal in the wings still, but that's not a JS
related framework.

On Thu, Nov 29, 2018 at 2:24 PM Shashi Prabhakar <notifications@github.com>
wrote:

> @dhh <https://github.com/dhh> keeping my eyes peeled for the promised new
> JS framework for Rails 6.0 reveal! (Haven't found anything satisfying so
> far with either React, or Stimulus or Ember - hope its a new "Rails Way"
> framework)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/rails/rails/pull/33079#issuecomment-443016183>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAAKtSfFeAoI1p87DRkq8ikb7a14INOPks5u0F6LgaJpZM4UegdM>
> .
>
